### PR TITLE
implement history service and controller

### PR DESCRIPTION
**Pull request by vlad12345123**

### CODE REVIEW

File: exalearn-api/ExaLearn.Bl/DTO/HistoryDTO.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using ExaLearn.Dal.Entities;
+using System;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class HistoryDTO
+    {
+        public int Id { get; set; }
+
+        public User User { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/HistoryDTO.cs]: UserDTO, n-layer architecture

File: exalearn-api/ExaLearn.Bl/DTO/HistoryDTO.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+        public Test Test { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/HistoryDTO.cs]: TestDTO

File: exalearn-api/ExaLearn.Bl/Interfaces/IHistoryService.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace ExaLearn.Bl.Interfaces
+{
+    public interface IHistoryService
+    {
+        public Task<HistoryDTO> AddAsync(HistoryDTO userDTO);
+
+        public Task<HistoryDTO> GetAsync(int id);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Interfaces/IHistoryService.cs]: GetByIdAsync

File: exalearn-api/ExaLearn.WebApi/Controllers/HistoryController.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿using ExaLearn.Bl.DTO;
+using ExaLearn.Bl.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+
+namespace ExaLearn.WebApi.Controllers
+{
+    [Route("api/users")]
+    [ApiController]
+    public class HistoryController : ControllerBase
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/HistoryController.cs:12]: [Authorize]

Comment (romanstat) [reply exalearn-api/ExaLearn.WebApi/Controllers/HistoryController.cs:12]: remove Async in naming only in controllers

File: exalearn-api/ExaLearn.Bl/DTO/HistoryDTO.cs (hunk 2 above)
Comment (vlad12345123) [reply exalearn-api/ExaLearn.Bl/DTO/HistoryDTO.cs]: TestDTO does not exist yet

File: exalearn-api/ExaLearn.WebApi/Controllers/HistoryController.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetHistoryById(int id)
+        {
+            return Ok(await _historyService.GetByIdAsync(id));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllHistoryAsync()
```
Comment (romanstat) [inline exalearn-api/ExaLearn.WebApi/Controllers/HistoryController.cs]: remove async

File: exalearn-api/ExaLearn.Bl/DTO/TestDTO.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+using Shared.Enums;
+using System;
+
+namespace ExaLearn.Bl.DTO
+{
+    public class TestDTO
+    {
+        public int Id { get; set; }
+
+        public User User { get; set; }
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/DTO/TestDTO.cs]: dto

**Review by romanstat — DISMISSED** [review]

**Review by romanstat — APPROVED** [review]

**Review by artemgael96 — APPROVED** [review]

**Pull request merged**